modules/luci-mod-network/htdocs/luci-static/resources/tools/network.js

```diff
@@ -519,7 +520,389 @@ return baseclass.extend({
+		o.depends({'type': 'bonding', 'policy': 'balance-alb'});
+
+		o = this.replaceOption(s, 'devgeneral', form.Flag, 'dynamic_lb', _('Dynamic load balance'), _('distribute traffic according to port load'));
+		o.default = o.disabled;
+		o.depends({'type': 'bonding', 'policy': 'balance-tlb'});
+
+		o = this.replaceOption(s, 'devadvanced', form.Value, 'resend_igmp', _('IGMP reports'));
+		o.description = _('Specifies the number of IGMP membership reports to be issued after a failover event');
+		o.placeholder = '1';
+		o.datatype = 'range(0, 255)';
+		o.depends('type', 'bonding');
+
+		o = this.replaceOption(s, 'devadvanced', form.Value, 'num_peer_notif', _('Peer notifications'));
+		o.description = _('Specify the number of peer notifications to be issued after a failover event.');
+		o.placeholder = '1';
+		o.datatype = 'range(0, 255)';
+		o.depends('type', 'bonding');
+
+		//primary ???
+
+		o = this.replaceOption(s, 'devadvanced', form.ListValue, 'primary_reselect', _('Primary port reselection policy'));
+		o.default = '';
+		o.value('', _(''));
+		o.value('always', _('Always'));
+		o.value('better', _('Better'));
+		o.value('failure', _('Failure'));
+		o.cfgvalue = function(/* ... */) {
+			var val = form.ListValue.prototype.cfgvalue.apply(this, arguments);
+
+			switch (val || '') {
+			case 'always':
+			case '0':
+				return 'always';
+
+			case 'better':
+			case '1':
+				return 'better';
+
+			case 'failure':
+			case '2':
+				return 'failure';
+
+			default:
+				return 'always';
+			}
+		};
+		o.depends('type', 'bonding');
+
+		o = this.replaceOption(s, 'devadvanced', form.ListValue, 'failover_mac', _('MAC address selection policy'));
+		o.default = '';
+		o.value('none', _('none'));
+		o.value('active', _('Active'));
+		o.value('follow', _('Follow'));
+		o.cfgvalue = function(/* ... */) {
+			var val = form.ListValue.prototype.cfgvalue.apply(this, arguments);
+
+			switch (val || '') {
+			case 'none':
+			case '0':
+				return 'none';
+
+			case 'active':
+			case '1':
+				return 'active';
+
+			case 'follow':
+			case '2':
+				return 'follow';
+
+			default:
+				return 'none';
+			}
+		};
+		o.depends('type', 'bonding');
+
+		o = this.replaceOption(s, 'devadvanced', form.ListValue, 'monitor_mode', _('Link monitoring mode'));
+		o.default = '';
+		o.value('arp', _('ARP link monitoring'));
+		o.value('mii', _('MII link monitoring'));
+		o.cfgvalue = function(/* ... */) {
+			var val = form.ListValue.prototype.cfgvalue.apply(this, arguments);
+
+			switch (val || '') {
+			case 'arp':
+			case '1':
+				return 'arp';
+
+			case 'mii':
+			case '2':
+				return 'mii';
+
+			default:
+				return 'mii';
+			}
+		};
+		o.depends('type', 'bonding');
+
+		// monitor_interval
+		o = this.replaceOption(s, 'devadvanced', form.Value, 'monitor_interval', _('Monitor Interval'));
+		o.description = _('Specifies the link monitoring frequency in milliseconds');
+		o.placeholder = '100';
+		// FIXME - we have to put a limit/range here somehow?
```

Comment:
Example
```js
		o.datatype = 'range(0,500)';
```

Reply:
I know that I can use the `range `method to specify the range as I already did on other controls, but the limit **might** depends on what you have selected as the preferred monitor system.
As the same value is used to configure MII or ARP monitoring.

See for `arp_interval` and `miimon` options in:
https://www.kernel.org/doc/Documentation/networking/bonding.txt

Reply:
What's sensible and what's absurd here? If they depend on others, then o.datatype is inflexible. Using `o.validate` allows access to other parameters already set via e.g.
```js
var value = this ? this.section.formvalue('devadvanced', 'xyz'): 1000
```
Then.... decide based on those other values, or what's set.